lib/internal/navigator.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const {
+  ObjectDefineProperties,
+  ReflectConstruct,
+} = primordials;
+
+const {
+  ERR_ILLEGAL_CONSTRUCTOR,
+} = require('internal/errors').codes;
+
+const {
+  kEnumerableProperty,
+} = require('internal/util');
+
+const {
+  getAvailableParallelism,
+} = internalBinding('os');
+
+class Navigator {
+  // Private properties are used to avoid brand validations.
+  #availableParallelism;
+
+  constructor() {
+    throw ERR_ILLEGAL_CONSTRUCTOR();
+  }
+
+  /**
+   * @return {number}
+   */
+  get hardwareConcurrency() {
+    this.#availableParallelism ??= getAvailableParallelism();
+    return this.#availableParallelism;
+  }
+}
+
+ObjectDefineProperties(Navigator.prototype, {
+  hardwareConcurrency: kEnumerableProperty,
+});
+
+module.exports = {
+  navigator: ReflectConstruct(function() {}, [], Navigator),
```

Comment:
create a private symbol instead of using ReflectConstruct. This will fix the build errors. In undici we do something along the lines of:

```js
const kInit = Symbol('blah')

class Navigator {
  constructor () {
    if (arguments[0] === kInit) { // arguments or a rest arg need to be used here so function.length is 0
      return
    }
    
    throw new TypeError('illegal constructor')
  }
}
```